### PR TITLE
cylc show: ignore past task prereqs

### LINKED ISSUE #4036: compute prerequisites for tasks outside n=0
See https://github.com/cylc/cylc-flow/pull/3938#issuecomment-752652607

Tasks that are not active are listed as having no prerequisites. This is especially unhelpful as we are not especially interested in the prerequisites of an active task.

Suggest computing prerequisites for the entire scheduler window (n=1 by default).

### PR DESCRIPTION
**Pull request by hjoliver**
These changes close #4036 

`cylc show` for past tasks loaded from the DB currently shows (incorrectly) unsatisfied prerequisites.

It might occasionally be useful to see which of the prerequisites of a conditionally-triggered past task led to it triggering, but we can't do that without creating another DB history table. (DB refactor might solve this in the future: https://github.com/cylc/cylc-flow/issues/3872#issuecomment-1125898049)

For now, the simplest solution is for clients to ignore the prerequisites of past tasks, which is not entirely unreasonable given the purpose of task prerequisites. (Ignoring them server-side isn't trivial, without causing trouble when past tasks are retriggered).

See also: https://github.com/cylc/cylc-ui/issues/1026

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.

- [x] Already covered by existing tests.

- [x] Appropriate change log entry included.

- [x] No documentation update required.

### CODE REVIEW

**Comment by hjoliver** [comment]
Also tidies `cylc show` output a bit.  Result for the example [here](https://github.com/cylc/cylc-flow/issues/4036#issuecomment-1125705824):

Screenshot from 2022-05-16 16-29-33

**Comment by hjoliver** [comment]
(rc3 - because the linked issue is against that, and there's a slim chance this could get merged in time).

File: tests/functional/cylc-show/06-past-present-future.t [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+TEST_NAME="${TEST_NAME_BASE}-validate"
+run_ok "${TEST_NAME}" cylc validate "${WORKFLOW_NAME}"
+
+# play workflow, shutdown and restart at task c so that task b (in n=1) gets
+# loaded from the DB.
+cylc play "${WORKFLOW_NAME}"
+cylc workflow-state -t c -p 1 --max-polls=10 --interval=2 --status=running \
+   "${WORKFLOW_NAME}"
+cylc stop --now --max-polls=10 --interval=2 "${WORKFLOW_NAME}"
```
Comment (MetRonnie) [inline tests/functional/cylc-show/06-past-present-future.t:35]: This test takes quite a while to run, could it be quickened by getting `c` to stop the workflow itself?

**Review by MetRonnie — APPROVED** [review]

File: cylc/flow/scripts/show.py [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
 """cylc show [OPTIONS] ARGS
 
-Display workflow and task information.
+Display workflow and task information, for tasks in the datastore window.
```
[the remark this answers is not in the document]
Comment (hjoliver) [reply cylc/flow/scripts/show.py:21]: It's tasks in the current n-window (i.e. n=0 and n=1, by default).  We need a term for the "datastore window".  Maybe "data window" or "graph window" or "n-window".  I'll go with the latter for now, and flag this as a discussion point.

File: tests/functional/cylc-show/06-past-present-future.t (hunk 1 above)
Comment (hjoliver) [reply tests/functional/cylc-show/06-past-present-future.t]: Done.  This is one of those tests that a DB-faking utility would be good for.

**Review by MetRonnie — APPROVED** [review]

File: cylc/flow/scripts/show.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MetRonnie) [reply cylc/flow/scripts/show.py]: I am happy with n-window
https://cylc.github.io/cylc-doc/nightly/html/glossary.html#term-window

**Comment by hjoliver** [comment]
Thanks @MetRonnie .  I'll merge this with one approval to get it in rc3 (it is only a peripheral command, and it is well tested).

**Pull request merged**